tests/e2e/tests/devfiles/NodeJS.spec.ts

```diff
@@ -0,0 +1,58 @@
+/*********************************************************************
+ * Copyright (c) 2019 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+import { WorkspaceNameHandler } from '../..';
+import 'reflect-metadata';
+import * as codeExecutionHelper from '../../testsLibrary/CodeExecutionTests';
+import * as commonLsTests from '../../testsLibrary/LsTests';
+import * as workspaceHandler from '../../testsLibrary/WorksapceHandlingTests';
+import * as projectManager from '../../testsLibrary/ProjectAndFileTests';
+
+const workspaceStack: string = 'NodeJS Express Web Application';
+const workspaceSampleName: string = 'nodejs-web-app';
+const workspaceRootFolderName: string = 'app';
+const fileFolderPath: string = `${workspaceSampleName}/${workspaceRootFolderName}`;
+const fileName: string = `app.js`;
+
+const taskDownloadDependencies: string = 'download dependencies';
+const taskRunWebApp: string = 'run the web app';
+const taskExpectedDialogText: string = 'A process is now listening on port 3000';
+
+suite(`${workspaceStack} test`, async () => {
+    suite(`Create ${workspaceStack}`, async () => {
+        workspaceHandler.createAndOpenWorkspace(workspaceStack);
+        projectManager.waitWorkspaceReadiness(workspaceSampleName, workspaceRootFolderName);
+    });
+    suite('Download dependencies', async () => {
+        codeExecutionHelper.runTask(taskDownloadDependencies, 60_000);
+        codeExecutionHelper.closeTerminal(taskDownloadDependencies);
+    });
+    suite(`'Language server validation'`, async () => {
+        projectManager.openFile(fileFolderPath, fileName);
+        commonLsTests.suggestionInvoking(fileName, 15, 20, 'require');
+        commonLsTests.autocomplete(fileName, 15, 20, 'require');
+        // commonLsTests.errorHighlighting(fileName, 'error_text;\n', 17); // error highlighting doesn't work in nodejs devfile https://github.com/eclipse/che/issues/16993
```

Comment:
Maybe better to remove commented code?

Reply:
it will be only commented temporarily, with a related issue link. once that's fixed the code is fully valid and the comment will be removed.

Reply:
I see, thank you